acvm-repo/brillig_vm/src/lib.rs

```diff
@@ -127,6 +131,14 @@ impl<'bb_solver, B: BlackBoxFunctionSolver> VM<'bb_solver, B> {
         self.status(VMStatus::ForeignCallWait { function, inputs })
     }
 
+    pub fn resolve_foreign_call(&mut self, foreign_call_result: ForeignCallResult) {
+        if self.foreign_call_counter < self.foreign_call_results.len() {
+            panic!("No unresolved foreign calls");
+        }
+        self.foreign_call_results.push(foreign_call_result);
+        self.status(VMStatus::InProgress);
+    }
```

Comment:
I think there's an argument to be made to remove the `foreign_call_results` argument from `VM::new` once we have this function.

Reply:
Agreed :+1: We should remove the check in that case, to allow feeding future results before the call opcodes are executed.